Sources/SwiftParser/Specifiers.swift

```diff
@@ -774,7 +766,7 @@ extension Parser {
       } else if let (spec, handle, matchedSubset) = self.at(anyIn: S.MisspelledThrowsTokenKinds.self, or: S.CorrectThrowsTokenKinds.self) {
         let misspelledThrows = self.eat(handle)
         unexpected.append(misspelledThrows)
-        if effectSpecifiers?.throwsSpecifier == nil {
+        if effectSpecifiers?.throwsClause == nil {
           if matchedSubset == S.CorrectThrowsTokenKinds.self {
             synthesizedThrows = missingToken(spec)
```

Comment:
We should synthesize an entire throws clause including the thrown type here instead of just synthesizing the `throws` keyword. This should be reflected in the Fix-It for something like

```swift
func foo() -> throws(MyError) Int {}
```

(it would be good to add a test case for this)

Reply:
There's a bit more to do in this area. We're not doing a good job in general of moving the whole throws clause around.

Reply:
Could you file issues so that we don’t forget to do it after the proposal is accepted?

Reply:
Filed an issue at https://github.com/apple/swift-syntax/issues/2391 to cover this